src/core/grid_based_algorithms/lb.cpp

```diff
@@ -945,34 +961,36 @@ inline void lb_collide_stream() {
 #endif
 
   Lattice::index_t index = lblattice.halo_offset;
-  for (int z = 1; z <= lblattice.grid[2]; z++) {
-    for (int y = 1; y <= lblattice.grid[1]; y++) {
-      for (int x = 1; x <= lblattice.grid[0]; x++) {
+  for (int z = 0; z < lblattice.grid[2]; z++) {
```

Comment:
Please either change this back so that `x, y, z` corresponds to the correct index, or remove the altogether, I think a single loop would work too.